unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -3516,6 +3572,265 @@ bool includeInvisible
+                        # endif
+                        foreach (float x in xVals) {
+                            foreach (float y in yVals) {
+                                foreach (float z in zVals) {
+                                    Vector3 worldPoint = bc.transform.TransformPoint(bc.center + new Vector3(x, y, z));
+                                    /*
+                                    # if UNITY_EDITOR
+                                    if (lastPoint.HasValue) {
+                                        Debug.DrawLine(lastPoint.Value, worldPoint, Color.red, 10.0f);
+                                    } else {
+                                        lastPoint = worldPoint;
+                                    }
+                                    # endif
+                                    */
+                                    yield return worldPoint;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public void BBoxDistance(string objectId0, string objectId1, int divisions=3) {
+            SimObjPhysics sop0 = getSimObjectFromId(objectId0);
+            SimObjPhysics sop1 = getSimObjectFromId(objectId1);
+            if (sop0 == null || sop1 == null) {
+                actionFinishedEmit(false); // Error message set already by getSimObjectFromId
+                return;
+            }
+            sop0.syncBoundingBoxes(forceCreateObjectOrientedBoundingBox: true); // Ensures the sop has an object oriented bounding box attached
+            sop1.syncBoundingBoxes(forceCreateObjectOrientedBoundingBox: true);
+
+            BoxCollider c0 = sop0.BoundingBox.GetComponent<BoxCollider>();
+            BoxCollider c1 = sop1.BoundingBox.GetComponent<BoxCollider>();
+
+            float dist = float.PositiveInfinity;
+
+            c1.enabled = true;
+            foreach (Vector3 p in pointsOnSurfaceOfBoxCollider(c0, divisions)) {
+                Vector3 pLocal = c1.transform.InverseTransformPoint(p) - c1.center;
+                Vector3 size = c1.size;
+                if (
+                    (-0.5f * size.x < pLocal.x && pLocal.x < 0.5f * size.x) &&
+                    (-0.5f * size.y < pLocal.y && pLocal.y < 0.5f * size.y) &&
+                    (-0.5f * size.z < pLocal.z && pLocal.z < 0.5f * size.z)
+                ) {
+# if UNITY_EDITOR
+                    Debug.Log($"{objectId0} is inside {objectId1}, distance is 0");
+# endif
+                    c1.enabled = false;
+                    actionFinishedEmit(true, actionReturn: 0f);
+                    return;
+                }
+                Vector3 closestP = c1.ClosestPoint(p);
+# if UNITY_EDITOR
+                Debug.DrawLine(p, closestP, Color.red, 10.0f);
+# endif
+                dist = Mathf.Min(dist, Vector3.Distance(p, closestP));
+            }
+            c1.enabled = false;
+
+            c0.enabled = true;
```

Comment:
What does enabling/disabling each do?

Reply:
If the colliders aren't enabled when running functions that do physics checks (namely `ClosestPoint` in this code) then we will get incorrect results (this is a hard thing to debug as, iirc, it doesn't throw an error).